### PR TITLE
feat: add `--exclude` CLI flag

### LINKED ISSUE #4148: Allow to exclude test files with cli
### Clear and concise description of the problem

I have the following use case for an exclude flag in the vitest cli:

We have a codebase with some unit tests using vitest in `./src/`
Alongside this code base there are some integration tests sharing the dto interfaces from `./src/`. The integration tests are in `./tests/api/` and use their own vitest config in `./tests/api/vitest.config.ts`.

We use `lint-staged` to execute relevant tests with `vite related` like the [docs suggest](https://vitest.dev/guide/cli.html#vitest-related).

However - those integration tests should not run in our commit hook due to their fragile nature. So I'd like to pass an exclude flag like so `vitest related --exclude './tests/api/**'`.

I couldn't find a solution with the already available CLI options. `-t` will only work on test names, but I want to exlude *files*.

### Suggested solution

Introduce an `--exclude` that'll exclude all tests matching a regex pattern in this CLI run. It'll override any exclude/include pattern from the vitest config files.

### Alternative

_No response_

### Additional context

_No response_

### Validations

- [X] Follow our [Code of Conduct](https://github.com/vitest-dev/vitest/blob/main/CODE_OF_CONDUCT.md)
- [X] Read the [Contributing Guidelines](https://github.com/vitest-dev/vitest/blob/main/CONTRIBUTING.md).
- [X] Read the [docs](https://vitest.dev/guide/).
- [X] Check that there isn't already an issue that request the same feature to avoid creating a duplicate.

### PR DESCRIPTION
**Pull request by Namchee**
### Description

This pull request adds a new flag `--exclude` as a CLI option that allows users to specify extra glob patterns to be ignored from tests. By default, it will merge the globs with existing default excludes.

Closes #4148 

### Please don't delete this checklist! Before submitting the PR, please make sure you do the following:
- [x] It's really useful if your PR references an issue where it is discussed ahead of time. If the feature is substantial or introduces breaking changes without a discussion, PR might be closed.
- [x] Ideally, include a test that fails without this PR but passes with it.
- [x] Please, don't make changes to `pnpm-lock.yaml` unless you introduce a new test example.

### Tests
- [x] Run the tests with `pnpm test:ci`.

### Documentation
- [x] If you introduce new functionality, document it. You can run documentation with `pnpm run docs` command.

### Changesets
- [x] Changes in changelog are generated from PR name. Please, make sure that it explains your changes in an understandable manner. Please, prefix changeset messages with `feat:`, `fix:`, `perf:`, `docs:`, or `chore:`.

### CODE REVIEW

File: packages/vitest/src/node/cli.ts [hunk 1]
```diff
@@ -151,6 +153,9 @@ function normalizeCliOptions(argv: CliOptions): CliOptions {
   else
     delete argv.dir
 
+  if (argv.exclude)
+    argv.exclude = [...defaultExclude, ...toArray(argv.exclude)]
```
Comment (sheremet-va) [inline packages/vitest/src/node/cli.ts:157]: I don't think we should do that. We probably want to extend user's `exclude` pattern, not the default one. Maybe we can just move it into another field (`extraExclude`/`cliExclude`) and spread it in the `vitest/src/config.ts` file?

**Review by sheremet-va — CHANGES_REQUESTED** [review]
It would also be nice if you can provide a test case for this feature. I don't think we have tests specifically for CLI commands, but you can create one.

**Comment by Namchee** [comment]
Good idea, will create a test for CLIs later.

File: packages/vitest/src/node/cli.ts (hunk 1 above)
Comment (Namchee) [reply packages/vitest/src/node/cli.ts]: Solved by ae05f63

**Comment by Namchee** [comment]
@sheremet-va , I think I have difficulties creating the CLI test since I don't know how to access the config from the running `vitest` instance. Do you have a pointer regarding this?

**Comment by sheremet-va** [comment]
> I think I have difficulties creating the CLI test since I don't know how to access the config from the running vitest instance. Do you have a pointer regarding this?

You can see an example of a CLI test in #4399

File: packages/vitest/src/node/config.ts [hunk 2]
```diff
@@ -175,6 +175,9 @@ export function resolveConfig(
       resolved.server.deps![option] = resolved.deps[option] as any
   })
 
+  if (options.exclude)
+    resolved.exclude.push(...options.exclude)
```
Comment (sheremet-va) [inline packages/vitest/src/node/config.ts:179]: I don't think this works as you expect. On line 84 `resolved` is created - you can see that options from CLI will just override user config.

This is why I proposed storing the value in something like `cliExcludes` and remove `excludes` before merging into resolved

**Review by sheremet-va — CHANGES_REQUESTED** [review]

**Comment by Namchee** [comment]
It seems like the test utils cannot navigate to the test fixtures correctly. Will debug this issue later.

File: packages/vitest/src/node/config.ts (hunk 2 above)
Comment (Namchee) [reply packages/vitest/src/node/config.ts]: Solved by 9d382a3

**Review by sheremet-va — APPROVED** [review]

**Pull request merged**